### ISSUE TITLE
Do not export Private\*.ps1 functions

### ISSUE THREAD

**Issue by vercellone** [joel74/POSH-LTM-Rest#37]
Private\*.ps1 functions are being exported, but should not be.  Reference PR #36

**Comment by dsolodow** [comment]
I see two ways to accomplish this:
1) modify the .psm1 file only to export Public functions (http://ramblingcookiemonster.github.io/Building-A-PowerShell-Module/)
2) list the public functions in 'FunctionsToExport' in the module manifest. 

Option 1 has the advantage of being a one and done, whereas 2 would require a manifest update when the cmdlet list changes. 
Option 2 has the advantage of fixing auto-load; currently if you try say a **Get-F5Session** without first importing the module it fails; 2 would fix this.

**Comment by vercellone** [comment]
Great stuff from @RamblingCookieMonster as usual.

I like the one and done benefit of option 1, but option 2 is probably the better long term choice.

**Comment by dsolodow** [comment]
@joel74, which is your preference? I'm inclined to option 2 as well, but figured would ask as you're the core dev.
Once we know, I'll make the mods and a PR.

**Comment by joel74** [comment]
@dsolodow, my one concern with Option 2 is that it's a manual process. What happens if a function gets missed? How would we know without doing a visual compare? RamblingCookieMonster mentions  exporting $Public.Basename to dynamically get the list of public functions while the module is in active development, and it feels like that's where this module currently is. I could live with either option, though.

**Comment by RamblingCookieMonster** [comment]
You could theoretically have a build process (e.g. maybe hosted on AppVeyor) that reads your public functions and updates the psd1?  You would get the benefits of having the functions hard coded in the psd1, without having to manually update it.

Cheers!

**Comment by vercellone** [comment]
Perfect.  Best of both worlds.  I already have a draft appveyor.yml file to run PSScriptAnalyzer and publish the module based on http://ramblingcookiemonster.github.io/Building-A-PowerShell-Module/.  I'll incorporate this feature.

**Comment by vercellone** [comment]
@joel74 I can do a proof of concept on my fork, but ultimately you'd have to register and configure appveyor for this repository.  Do you foresee any problems with that?

**Comment by joel74** [comment]
I haven't used appveyor before, but I don't think it will be a problem to set it up.

**Comment by vercellone** [comment]
Look over [appveyor.yml](https://github.com/vercellone/POSH-LTM-Rest/blob/master/appveyor.yml) and the [build results](https://ci.appveyor.com/project/vercellone/posh-ltm-rest) and let me know what you think.  If you don't want to publish to powershellgallery, then I'll want to tweak the zip artifact contents so they don't include the .git\ folder nor the appveyor.yml file itself.

If you are willing to publish to [PowerShell Gallery](http://www.powershellgallery.com/) (and, i hope that you are), then I'd rather pursue that and forego the zip artifact altogether.

**Comment by joel74** [comment]
I've published to the gallery, not every commit but fairly regularly, so that's fine. I'd like it to be published automatically. Does appveyor need to hook up with the existing module in the gallery? I'll check out the appveyor stuff soon.

**Comment by vercellone** [comment]
[Publishing](https://www.powershellgallery.com/packages/upload) is as simple as:
```PowerShell
Publish-Module -Name 'F5-LTM Module' -NuGetApiKey '[Your API Key]'
```
1. [View your PowerShell Gallery profile]() to get your API Key.
1. [Encrypt](https://ci.appveyor.com/tools/encrypt) it, so you can safely include the encrypted value in the appveyor.yml file.

**Tip**: I highly recommend AppVeyor's [Validate YAML configuration](https://ci.appveyor.com/tools/validate-yaml) form.

**Comment by joel74** [comment]
Cool. So you're saying that if we publish to PSGallery, you'll remove the .7zip portion? 
I see where it's calling msiexec - what's the role of nuget.exe?

Does appveyor handle versioning? I know that PSGallery requires a version bump every time one publishes. The encrypted API key value is: 
secure: NvJlPynne4N7tJ/8DM2LFAKoZu5v9e35PvbAJNOCiXbXJRHrJ+OetTQsO/bYjNo5

**Comment by vercellone** [comment]
- Yes, either an appeyor artifact (zip file is just one option) or publish to PSGallery, not both.
- The msiexec install of PackageManagement_x64.msi should not be necessary when os: WMF 5 is specified.  But, I haven't verified that yet.
- Refer to the [Get Started with the PowerShell Gallery](http://www.powershellgallery.com/GettingStarted?section=Get%20Started) to learn more about Nuget's role.  Again, this might not be necessary for os: WMF 5, but I haven't verified.
- AppVeyor includes build numbering, and I've included setting the version in the psd1 in the same command used to set the FunctionsToExport.  You can download the [zip artifact](https://ci.appveyor.com/api/buildjobs/ws0kqhsv3m9rnq22/artifacts/F5-LTM.1.3.13.zip) to verify the result.
- I'm fairly certain the secure string is AppVeyor user specific.  But, I'll add it for the purpose of a simpler pull request.

**Comment by vercellone** [comment]
- PackageManagement_x64.msi is NOT necessary
- Get-PackageProvider -Name NuGet -Force IS necessary
- I moved the module contents into a folder named 'F5-LTM Module'.  That should simplify Publish-Module, but you may need to experiment with the naming to get the desired result in the gallery.

**Comment by joel74** [comment]
I'd like to move the module back into a folder named 'F5-LTM' as that will sync up with the published module in the PowerShell gallery. Any reason not to?

**Comment by vercellone** [comment]
Sounds good.

**Comment by vercellone** [comment]
dbb5410 of PR #43 fixes #40 

The delimiter was wrong, so all Functions were still be exported.  Fixing the delimiters should work.  After import of the latest module from PSGallery, the following command should not include the private functions:
```PowerShell
Get-Command | ? Source -eq 'f5-ltm' 
```

**Comment by vercellone** [comment]
This issue can be closed based on version [1.3.14](http://www.powershellgallery.com/packages/f5-ltm/1.3.14).

**Comment by joel74** [comment]
Confirmed
